generators/react/templates/src/main/webapp/app/shared/layout/header/header.tsx.ejs

```diff
@@ -52,8 +52,13 @@ export interface IHeaderProps {
 
 const Header = (props: IHeaderProps) => {
   const [menuOpen, setMenuOpen] = useState(false);
+
 <%_ if (enableI18nRTL) { _%>
-  useEffect(() => document.querySelector('html').setAttribute('dir', isRTL(Storage.session.get('locale')) ? 'rtl' : 'ltr'));
+  const currentLocale = useAppSelector(state => state.locale.currentLocale);
+
+  useEffect(() => {
+    document.querySelector('html').setAttribute('dir', isRTL(currentLocale) ? 'rtl' : 'ltr');
+  }, [currentLocale]);
```

Comment:
You are faster 😄.

I agree with you solution if the following are handled:
- the location looks wrong, since it will be more related to storage (generic solution) than it is related to the menu selection it should be in app.tsx.
- there is already a handler for locale rtl handling bellow, so the code will execute twice on each locale change.

Reply:
I implemented the suggestions but now saw your solution and using templating to set the default is quite elegant and better than what I proposed.

What I definitely think still makes sense to add is the dependency array for the useEffect. Otherwise, it will run on every re-render, which is even worse than twice per language change. :smile:

Moving the effect to app.tsx and setting the HTML attribute like I suggested could still be worthwhile.